generators/app/templates/src/redux/store.ejs

```diff
@@ -16,7 +18,28 @@ import drawer from './drawer/reducer';
 <%_ if(features.drawerios || features.drawerandroid || features.pushnotifications || features.login) { _%>
 
 <%_ } _%>
-const nav = (state, action) => Navigator.router.getStateForAction(action, state) || state;
+const nav = (state, action) => {
+  const nextState = Navigator.router.getStateForAction(action, state);
+  // Prevents navigating twice to the same route but allow navigate if same routes with different params
+  if (state && nextState) {
+    // Getting routes and params
+    const currentParams = Navigator.router.getPathAndParamsForState(state).params;
+    const nextParams = action.params;
+    const stateRouteName = getCurrentRouteName(state);
+    const nextStateRouteName = getCurrentRouteName(nextState);
+    // Check if has same params
+    const hasSameParams = isEqual(currentParams, nextParams);
```

Comment:
I'd check for `hasSameParams` inside the `if (hasSameRoute)` block so we avoid doing `isEqual` always.